geode/src/test/java/org/apache/calcite/adapter/geode/rel/GeodeZipsTest.java

```diff
@@ -184,7 +206,105 @@ public void testItemPredicate() {
             + "  GeodeProject(lat=[ITEM($2, 0)], lon=[ITEM($2, 1)])\n"
             + "    GeodeSort(fetch=[1])\n"
             + "      GeodeFilter(condition=[>(ITEM($2, 0), 0)])\n"
-            + "        GeodeTableScan(table=[[geode, zips]])\n");
+            + "        GeodeTableScan(table=[[geode, zips]])\n")
+        .queryContains(
+            GeodeAssertions.query("SELECT loc[0] AS lat, "
+            + "loc[1] AS lon FROM /zips WHERE loc[0] > 0 LIMIT 1"));
+  }
+
+  @Test
+  public void testWhereWithOrForStringField() {
+    String expectedQuery = "SELECT state AS state FROM /zips "
+        + "WHERE state IN SET('MA', 'RI')";
+    calciteAssert()
+        .query("SELECT state as state "
+            + "FROM view WHERE state = 'MA' OR state = 'RI'")
+        .returnsCount(6)
+        .queryContains(
+            GeodeAssertions.query(expectedQuery));
+  }
+
+  @Test
+  public void testWhereWithOrForNumericField() {
+    calciteAssert()
+        .query("SELECT pop as pop "
+            + "FROM view WHERE pop = 34035 OR pop = 40173")
+        .returnsCount(2)
+        .queryContains(
+            GeodeAssertions.query("SELECT pop AS pop FROM /zips WHERE pop IN SET(34035, 40173)"));
+  }
+
+  @Test
+  public void testWhereWithOrForNestedNumericField() {
+    String expectedQuery = "SELECT loc[1] AS lan FROM /zips "
+        + "WHERE loc[1] IN SET(43.218525, 44.098538)";
+
+    calciteAssert()
+        .query("SELECT loc[1] as lan "
+            + "FROM view WHERE loc[1] = 43.218525 OR loc[1] = 44.098538")
+        .returnsCount(2)
+        .queryContains(
+            GeodeAssertions.query(expectedQuery));
+  }
+
+  @Test
+  public void testWhereWithOrForLargeValueList() throws Exception {
+    Cache cache = POLICY.cache();
+    QueryService queryService = cache.getQueryService();
+    Query query = queryService.newQuery("select state as state from /zips");
+    SelectResults results = (SelectResults) query.execute();
+
+    Set<String> stateList = new LinkedHashSet<>();
```

Comment:
[SelectResults](https://geode.apache.org/releases/latest/javadoc/org/apache/geode/cache/query/SelectResults.html) already implements Collection so you can convert it to stream:
```java
Set<String> states = results.stream()
      .map(s -> (String) s.get("state"))
       .collect(Collectors.toCollection(LinkedHashSet::new))
```